src/Stratis.Features.Unity3dApi/Controllers/Unity3dController.cs

```diff
@@ -406,5 +441,101 @@ public TipModel GetTip()
                 return null;
             }
         }
+        
+        /// <summary>
+        /// Gets a smart contract transaction receipt. Receipts contain information about how a smart contract transaction was executed.
+        /// This includes the value returned from a smart contract call and how much gas was used.  
+        /// </summary> 
+        /// <param name="txHash">A hash of the smart contract transaction (the transaction ID).</param> 
+        /// <returns>The receipt for the smart contract.</returns> 
+        /// <response code="200">Returns transaction receipt</response>
+        /// <response code="400">Transaction not found</response>
+        [Route("api/[controller]/receipt")]
+        [HttpGet]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public ReceiptResponse GetReceiptAPI([FromQuery] string txHash)
+        {
+            ReceiptResponse receiptResponse;
+            uint256 txHashNum = new uint256(txHash);
+            Receipt receipt = this.receiptRepository.Retrieve(txHashNum);
+
+            if (receipt == null)
+                return null;
+
+            uint160 address = receipt.NewContractAddress ?? receipt.To;
+
+            if (!receipt.Logs.Any())
+            {
+                receiptResponse = new ReceiptResponse(receipt, new List<LogResponse>(), this.network);
+            }
+            else
+            {
+                var deserializer = new ApiLogDeserializer(this.primitiveSerializer, this.network, this.stateRoot, this.contractAssemblyCache);
+                List<LogResponse> logResponses = deserializer.MapLogResponses(receipt.Logs);
+
+                receiptResponse = new ReceiptResponse(receipt, logResponses, this.network);
+            }
+            
+            return receiptResponse;
```

Comment:
Looks good Ivan, I would just add this controller method and the one below in a try catch, like we do in all the other controller methods. Just so that there is some consistency? WDYT?

Example:

https://github.com/stratisproject/StratisFullNode/blob/master/src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs#L100

Reply:
There is no need for try catch since if there is an exception it will be seen in node's logs. 
Also we can't use same error handling since controller's methods return explicit types, not json like our other controllers.